src/main/java/uk/gov/hmcts/cmc/claimstore/utils/DirectionsQuestionnaireUtils.java

```diff
@@ -51,11 +51,15 @@ private static String getDefendantHearingCourt(Response defendantResponse) {
         if (defendantResponse.getResponseType() == FULL_DEFENCE) {
             return ((FullDefenceResponse) defendantResponse).getDirectionsQuestionnaire()
                 .orElseThrow(IllegalStateException::new)
```

Comment:
All of those can be simplified as 

```((PartAdmissionResponse) defendantResponse).getDirectionsQuestionnaire()
                .flatMap(DirectionsQuestionnaire::getHearingLocation)
                .map(HearingLocation::getCourtName)
                .orElseThrow(IllegalStateException::new);``` 
up to you if you want to do it, I feel having an exception at the end is usually more readable

Reply:
done